### ISSUE TITLE
区间柱图设置了barMinHeight或者label为outside后， 拖动缩略轴，不在时间范围内的数据不会在图表中隐藏

### ISSUE THREAD

**Issue by H-Sven** [VisActor/VChart#3191]
<img width="1628" alt="image" src="https://github.com/user-attachments/assets/951531c7-263f-4cb7-82bd-9d9cc5b6d0e7">
https://www.visactor.io/vchart/demo/data-zoom/state-with-data?keyword=dataZoom

**Comment by H-Sven** [comment]
这个问题也是比较急需解决的

**Comment by H-Sven** [comment]
@skie1997  这个问题能尽快排查一下不

**Comment by H-Sven** [comment]
这个问题挺着急的， 有办法解决吗

**Comment by skie1997** [comment]
> <img alt="image" width="1628" src="https://private-user-images.githubusercontent.com/26042365/364761140-951531c7-263f-4cb7-82bd-9d9cc5b6d0e7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MDMwNjUsIm5iZiI6MTcyNTYwMjc2NSwicGF0aCI6Ii8yNjA0MjM2NS8zNjQ3NjExNDAtOTUxNTMxYzctMjYzZi00Y2I3LTgyYmQtOWQ5Y2M1YjZkMGU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA2VDA2MDYwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlMWY5MDk1ZDQ0MWZhMDAwYTE4MjNjNDUyNzRiNmQwZTI5YjRlNDU2NDVhOTEzYjI5NDNiYjJjMzk4MTVkZDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tft5bb3lrSOur4yxxlkFeDB8x9_zAU0QeqxnkYIlA1Y"> https://www.visactor.io/vchart/demo/data-zoom/state-with-data?keyword=dataZoom

在条形图中，barMinWidth对应的是柱子高度，barMinHeight对应的是柱子宽度，所以设置了barMinHeight，柱子的宽度最小为30.  在这个case中，应该设置barMinWidth比较合理
![image](https://github.com/user-attachments/assets/3fb6ac58-2d8c-4041-b76f-a9b1a3d957c2)

**Comment by H-Sven** [comment]
> > <img alt="image" width="1628" src="https://private-user-images.githubusercontent.com/26042365/364761140-951531c7-263f-4cb7-82bd-9d9cc5b6d0e7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MDMwNjUsIm5iZiI6MTcyNTYwMjc2NSwicGF0aCI6Ii8yNjA0MjM2NS8zNjQ3NjExNDAtOTUxNTMxYzctMjYzZi00Y2I3LTgyYmQtOWQ5Y2M1YjZkMGU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA2VDA2MDYwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlMWY5MDk1ZDQ0MWZhMDAwYTE4MjNjNDUyNzRiNmQwZTI5YjRlNDU2NDVhOTEzYjI5NDNiYjJjMzk4MTVkZDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tft5bb3lrSOur4yxxlkFeDB8x9_zAU0QeqxnkYIlA1Y"> https://www.visactor.io/vchart/demo/data-zoom/state-with-data?keyword=dataZoom
> 
> 在条形图中，barMinWidth对应的是柱子高度，barMinHeight对应的是柱子宽度，所以设置了barMinHeight，柱子的宽度最小为30. 在这个case中，应该设置barMinWidth比较合理。 标签的问题还有待排查 @H-Sven ![image](https://private-user-images.githubusercontent.com/40675330/365054477-3fb6ac58-2d8c-4041-b76f-a9b1a3d957c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MDQwNDksIm5iZiI6MTcyNTYwMzc0OSwicGF0aCI6Ii80MDY3NTMzMC8zNjUwNTQ0NzctM2ZiNmFjNTgtMmQ4Yy00MDQxLWI3NmYtYTliMWEzZDk1N2MyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA2VDA2MjIyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwOTJhZGJjNzhlOWRhNmNmNmYyYjE2MjU3YTc3ZDNhYmY2ZDcxMDkxNmJlMmMxODVjZGZkYjUzNGNmYzVmM2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.F7pjMTgg2rqpbUX2KhYknWqoZP6QWA4kEckbG9Kv60A)

我知道barMinHeight对应的是柱子的宽度， 因为在区间柱图里面如果时间范围太小的话，可能看不到柱子，所以我通过barMinHeight设置了最小宽度。 

目前问题就是，我设置了barMinHeight 或者 label后， 拖动datazoom， 这条柱子无法隐藏。

预期的结果是能够正常隐藏掉

**Comment by H-Sven** [comment]
@skie1997  哈喽，请问这个问题还在排查吗

**Comment by H-Sven** [comment]
@skie1997  哈喽，这个问题有修复计划吗

**Comment by SafeWinter** [comment]
同意 skie1997 的意见。区间距离过小而强行指定最小高度（水平方向），这意味着图表所有图元都要按比例缩放，与其寄希望于改变滑块来实现缩放浏览，还不如将数据集改造成放大版。水平滑块按原设计本就不具备你想要的那种缩放效果。

**Comment by H-Sven** [comment]
> 与其寄希望于改变滑块来实现缩放浏览，还不如将数据集改造成放大版。

你的意思是将我的数据整体放大n倍吗？ 将每一条数据值*n ?

**Comment by SafeWinter** [comment]
> 你的意思是将我的数据整体放大n倍吗？ 将每一条数据值*n ?

我大概懂你意思了，设置 barMinHeight=30 后，其实已经让条形图每个矩形条水平方向的最小长度变为 30 了，问题就出在拖动缩放条时，本来不在视野内的矩形条也按 30 的长度继续显示，并且残留在窗口的左边（按理应该看不到才是）。

**Comment by H-Sven** [comment]
> > 你的意思是将我的数据整体放大n倍吗？ 将每一条数据值*n ?
> 
> 我大概懂你意思了，设置 barMinHeight=30 后，其实已经让条形图每个矩形条水平方向的最小长度变为 30 了，问题就出在拖动缩放条时，本来不在视野内的矩形条也按 30 的长度继续显示，并且残留在窗口的左边（按理应该看不到才是）。

对，就是这个问题。 但是作者好像没有打算修复的意思。 一直没有回复

**Comment by H-Sven** [comment]
@xile611 @skie1997  可以确认一下这是不是一个bug吗
